_gsocblogs/2023/blog_FASER_SumalyoDatta.md

```diff
@@ -0,0 +1,243 @@
+Code for the new Dockerfile can be found here : [sumalyo_dev branch on faser-docker](https://gitlab.cern.ch/faser/docker/-/tree/sumalyo_dev?ref_type=heads)
+<br>
+Merge Request: [MR #2](https://gitlab.cern.ch/faser/docker/-/merge_requests/2)<br>
+![MR_Merged](https://img.shields.io/badge/MR-Merged-green?style=for-the-badge&logo=appveyor)
+
+## Raw File Compressor
+In the first phase of the project, a standalone utility application `eventCompress` [Link to Code](https://gitlab.cern.ch/faser/faser-common/-/blob/compression_app/EventFormats/apps/eventCompress.cxx) was developed which can be used to compress existing raw data files and record performance metrics during the process. It can also do decompression tests and record metrics like decompression speed.
+```
+Usage: eventCompress [-n nEventsMax -l -d -s -o <output_file> -c <config_file> ] <input_raw_filename> 
+   -n <no. events>:   Run Compression for n events only (optional)
+   -l --enableLog:    Enable logging for metrics and benchmarks (optional) 
+   -d --decomress:    Run Decompression tests (optional)
+   -s --silent:       suppress logs (optional)
+   -o --write:        specify file to write out to
+   -c --config:       specify file to read compressor config
+```
+The idea here is to compress event-by-event reading from a raw file and alter the header information to indicate compression and specify the compression algorithm used. As each event was compressed individually, the time taken for compression and the ratio of the uncompressed size to the compressed size are recorded for each event. These are logged in a JSON file, which can be analysed with a Python notebook to visualise the data as graphs and to understand performance tradeoffs. You can view a sample report [here](https://cloud.datapane.com/reports/VkG8X2A/compression-report-an-overview/). 
+<br>
+Merge Request: [MR #50](https://gitlab.cern.ch/faser/faser-common/-/merge_requests/50)<br>
+![MR_Merged](https://img.shields.io/badge/MR-Merged-green?style=for-the-badge&logo=appveyor)
+<br>
+The logs and the python-based analysis notebooks can be found at [this repo](https://gitlab.cern.ch/faser/online/compression-studies). Please refer to the README for additional information.
+
+### The Best Candidate
+| ![Chart](https://sumalyo.github.io/benchmarkFASER/FASERImages/FASER-Report-Compression.jpg) | 
+|:--:| 
+| *Fig 2 Average Compression Ratio vs Average Compression Speed* |
+
+The approach for finding the best compressor is described below
+- Events were divided into a set of 10 classes based on event size (Class 0 having events of smallest size)
+- For each event class, the average compression speed and compression ratio were calculated (for each compressor)
+- The resulting points were plotted on a graph, and this helped to visualize the tradeoff. The compressor configuration offering the highest average compression ratio at the highest compression speed was considered optimal.
```

Comment:
A bit nitpicky since I know what you mean: Since there is a trade-off between compression ratio and speed, we will not get an algorithm that will achieve _both_ highest compression _and_ highest speed. I would rephrase by referring to your "acceptance region" in fig 2 and something like "The compressor configuration configuration offering the highest compression speed at an acceptable compression ratio" (or vice versa).